smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/SendingKafkaMessage.java

```diff
@@ -65,10 +65,15 @@ public MessageHeaders getHeaders() {
     }
 
     @Override
-    public Integer getPartition() {
-        if (partition == null || partition < 0) {
-            return null;
+    public int getPartition() {
+        if (partition < 0) {
+            return -1;
         }
         return partition;
     }
+
+    @Override
+    public long getOffset() {
+        return -1;
```

Comment:
Is the offset meaningless when sending, that's why we return -1?

Reply:
yes, the offset is the position of the record in the partition. It only makes sense when receiving a record.